src/components/TradeWidget/TokenRow.tsx

```diff
@@ -176,6 +181,19 @@ const TokenRow: React.FC<Props> = ({
     [inputId, setValue],
   )
 
+  const onKeyPress = useCallback(
+    (event: React.KeyboardEvent<HTMLInputElement>): void => {
+      if (event.key === 'Enter') {
+        // On enter, submit the form
+        handleSubmit(onSubmit)
```

Comment:
Forms submit on ENTER natively. Remove the whole onSubmit handling in TokenRow
What is useful to do on ENTER is to `removeExcessZeros(event)`

Reply:
I did all this mess because enter wasn't submitting the form. With your comment I realized it was because I was capturing enter with `preventInvalidChars`. Reverted this change and addressed as you suggested

Reply:
:+1: